test/test_extended_models.py

```diff
@@ -103,24 +103,97 @@ def test_weights_deserializable(name):
+    ]
+
+
 @pytest.mark.parametrize(
     "module", [models, models.detection, models.quantization, models.segmentation, models.video, models.optical_flow]
 )
 def test_list_models(module):
-    def get_models_from_module(module):
-        return [
-            v.__name__
-            for k, v in module.__dict__.items()
-            if callable(v) and k[0].islower() and k[0] != "_" and k not in models._api.__all__
-        ]
-
     a = set(get_models_from_module(module))
     b = set(x.replace("quantized_", "") for x in models.list_models(module))
 
     assert len(b) > 0
     assert a == b
 
 
+@pytest.mark.parametrize(
+    "include_filters",
+    [
+        None,
+        [],
+        (),
+        {},
+        "*resnet*",
+        ["*alexnet*"],
+        "*not-existing-model-for-test?",
+        ["*resnet*", "*alexnet*"],
+        ["*resnet*", "*alexnet*", "*not-existing-model-for-test?"],
+        ("*resnet*", "*alexnet*"),
+        set(["*resnet*", "*alexnet*"]),
+        {"*resnet*": "test", "*alexnet*": "test"},
+    ],
+)
+@pytest.mark.parametrize(
+    "exclude_filters",
+    [
+        None,
+        [],
+        (),
+        {},
+        "*resnet*",
+        ["*alexnet*"],
+        ["*not-existing-model-for-test?"],
+        ["resnet34", "*not-existing-model-for-test?"],
+        ["resnet34", "*resnet1*"],
+        ("resnet34", "*resnet1*"),
+        set(["resnet34", "*resnet1*"]),
+        {"resnet34": "test", "**resnet1**": "test"},
+    ],
+)
+def test_list_models_filters(include_filters, exclude_filters):
+    module = models
+    b = set(
+        x.replace("quantized_", "")
+        for x in models.list_models(module, include=include_filters, exclude=exclude_filters)
+    )
+    # Test include filters
+    # string filter
+    if isinstance(include_filters, str):
+        include_filters_base = include_filters.strip("*?")
+        a = set(x for x in get_models_from_module(module) if include_filters_base in x)
+    # No filter
+    elif include_filters is None:
+        a = set(x for x in get_models_from_module(module))
+    # Empty iterable
+    elif len(include_filters) == 0:
+        a = set(x for x in get_models_from_module(module))
+    # Iterable of filters
+    else:
+        a = set()
+        for include_f in include_filters:
+            include_filters_base = include_f.strip("*?")
+            a = a | set(x for x in get_models_from_module(module) if include_filters_base in x)
+
+    # Test exclude filters
+    if isinstance(exclude_filters, str):
+        exclude_filters_base = exclude_filters.strip("*?")
+        a_exclude = set(x for x in get_models_from_module(module) if exclude_filters_base in x)
+        a = a - a_exclude
+    elif exclude_filters is not None:
+        for exclude_f in exclude_filters:
+            exclude_filters_base = exclude_f.strip("*?")
+            a_exclude = set(x for x in get_models_from_module(module) if exclude_filters_base in x)
+            a = a - a_exclude
```

Comment:
Similar simplifications as above:

```suggestion
    if isinstance(exclude_filters, str):
        exclude_filters = [exclude_filters]
    if exclude_filters:
        for exclude_f in exclude_filters:
            exclude_filters_base = exclude_f.strip("*?")
            a_exclude = set(x for x in get_models_from_module(module) if exclude_filters_base in x)
            a = a - a_exclude
```